lib/runner/concurrency/index.js

```diff
@@ -91,7 +92,11 @@ class Concurrency extends EventEmitter {
    */
   runTestEnvironments(envs, args, availColors) {
     return Promise.all(envs.map((environment, index) => {
-      let childProcess = this.createChildProcess(environment, args, ['--env', environment], index);
+      const extraArgs = ['--env', environment];
+      if (this.isSafariEnvPresent) {
```

Comment:
Why can't we run in serial mode only for safari and let chrome sessions run in parallel?

Reply:
When we run safari sessions in serial and remaining env sessions in parallel, The reporter is showing results from only safari session (or session which was run in serial). Since we are creating child processes for each chrome sessions (to run in parallel), the master process is not receiving the results of chrome env (parallel)  and only receiving from safari env (serial).
cc @swrdfish

Reply:
Shouldn't we fix this?

Reply:
We should fix this in the worker thread implementation. Can merge this one for now.